LTO bug in metadata-test

Ran into a weird bug where enabling LTO with a release build on GCC 14.2 would error out. The bug did not appear when I tested with Clang. Its a 1-line fix so I'm putting in a PR.

```sh
$ gcc --version
gcc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7)
```

```sh
c++  -o test/metadata-test test/metadata-test.p/metadata-test.cpp.o test/metadata-test.p/.._src_metadata.cpp.o -Werror -flto -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 '-Wl,-rpath,/usr/local/lib64:$ORIGIN' -Wl,-rpath-link,/usr/local/lib64 -Wl,--start-group /usr/lib64/libgtest_main.so /usr/lib64/libgtest.so /usr/local/lib64/libzim.so /usr/lib64/libicui18n.so /usr/lib64/libicuuc.so /usr/lib64/libicudata.so /usr/lib64/libdocopt.so /usr/lib64/libgumbo.so /usr/lib64/libmagic.so /usr/lib64/libz.so -Wl,--end-group
In function ‘copy’,
    inlined from ‘__ct ’ at /usr/include/c++/14/bits/basic_string.h:688:23,
    inlined from ‘operator+’ at /usr/include/c++/14/bits/basic_string.h:3721:46,
    inlined from ‘fakePNG’ at ../test/metadata-test.cpp:7:52:
/usr/include/c++/14/bits/char_traits.h:427:56: error: ‘__builtin_memcpy’ reading 109 bytes from a region of size 16 [-Werror=stringop-overread]
  427 |         return static_cast<char_type*>(__builtin_memcpy(__s1, __s2, __n));
      |                                                        ^
../test/metadata-test.cpp: In function ‘fakePNG’:
../test/metadata-test.cpp:7:52: note: at offset 16 into source object ‘<anonymous>’ of size 32
    7 |   return "\x89PNG\r\n\x1a\n" + std::string(100, 'x');
      |                                                    ^
lto1: all warnings being treated as errors
make: *** [/tmp/ccxjkOQs.mk:2: /tmp/ccmiKevA.ltrans0.ltrans.o] Error 1
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
```

--- a/test/metadata-test.cpp
+++ b/test/metadata-test.cpp
@@ -4,7 +4,7 @@
 
 std::string fakePNG()
 {
-  return "\x89PNG\r\n\x1a\n" + std::string(100, 'x');
+  return std::string{"\x89PNG\r\n\x1a\n"} + std::string(100, 'x');
 }
 
 TEST(Metadata, isDefaultConstructible)

